core/src/main/java/cucumber/runtime/Runtime.java

```diff
@@ -200,7 +209,75 @@ public Runtime build() {
 
             final RerunFilters rerunFilters = new RerunFilters(this.runtimeOptions, featureLoader);
             final Filters filters = new Filters(this.runtimeOptions, rerunFilters);
-            return new Runtime(plugins, this.runtimeOptions, this.eventBus, filters, runnerSupplier, featureSupplier);
+            return new Runtime(plugins, this.runtimeOptions, this.eventBus, filters, runnerSupplier, featureSupplier, executor);
+        }
+    }
+
+    private static final class SameThreadExecutorService implements ExecutorService {
```

Comment:
could extends `AbstractExecutorService` would reduce the number of methods to implement without change to what you need, other than the other `submit` and `invoke` methods being pre-supported

Reply:
Yup.